shotover-proxy/src/transforms/protect/mod.rs

```diff
@@ -30,6 +33,161 @@ pub struct Protect {
     key_id: String,
 }
 
+impl Protect {
+    /// encodes a Protected object into a byte array.  This is here to centeralize the serde for
+    /// the Protected object.
+    /// Returns an error if a Plaintext Protected object is passed
+    fn encode(protected: &Protected) -> Result<Vec<u8>> {
+        match protected {
+            Protected::Plaintext(_) => Err(anyhow!("can not encode plain text")),
+            Protected::Ciphertext { .. } => Ok(serde_json::to_vec(protected)?),
+        }
+    }
+
+    /// decodes a byte array into the Protected object.  This is here to centeralize the serde for
+    /// the Protected object.
+    fn decode(data: &[u8]) -> Result<Protected> {
+        Ok(serde_json::from_slice(data)?)
+    }
+
+    /// get the list of protected columns for the specified table name.  Will return `None` if no columns
+    /// are defined for the table.
+    fn get_protected_columns(&self, table_name: &FQName) -> Option<&Vec<String>> {
+        // TODO replace "" with cached keyspace name
+        if let Some(tables) = self
+            .keyspace_table_columns
+            .get(table_name.extract_keyspace(""))
+        {
+            tables.get(&table_name.name)
+        } else {
+            None
+        }
+    }
+
+    /// extractes the protected object from the message value.  Resulting object is a Protected::Ciphertext
+    fn extract_protected(&self, value: &MessageValue) -> Result<Protected> {
+        match value {
+            MessageValue::Bytes(b) => Protect::decode(&b[..]),
+            MessageValue::Varchar(hex_value) => {
```

Comment:
We never create a MessageValue::Varchar (only a MessageValue::Bytes) so we should never attempt to decode it.